src/main/java/cubicchunks/lighting/LightingManager.java

```diff
@@ -46,7 +46,8 @@ public LightingManager(ICubicWorld world) {
 
 		this.skylightCubeDiffuseProcessor = new SkyLightCubeDiffuseProcessor(world, "Sky Light Diffuse", 5);
 
-		this.firstLightProcessor = new FirstLightProcessor("First Light", world.getCubeCache(), 1);
+		// TODO: Setting GeneratorStage to null. Analyze effects.
+		this.firstLightProcessor = new FirstLightProcessor(null, "First Light", world.getCubeCache(), 1);
```

Comment:
The state handling for popuation is just more weird than normal. It's non-linear because populating some cube will alter state of other cubes. And while lighting is needed for population, I think generation pipeline should be flexible enough to allow that. So basically something like a branching generation pipeline.

Lighting here is done in a bit weird way, you should really take a look at FirstLightProcessor. It basically updates lighting in that cube and all cubes below.

And when explaining how it works before I got a bit confused myself. Here the null value will circumvent checking chunk existance in case when queueFirstLightCalculation is called (honestly, I totally forgot that this part of code even exists). It will make the check almost always succeed, and lighting will be calculated when it shouldn't be.

And the change in isAreaLoaded is just breaking the whole FirstLightProcessor as a generation stage. When FirstLightProcessor is ran, the cube is in LIGHTING stage. And FirstLightProcessor calls World.checkLightFor. Which ises isAreaLoaded. Which will return false because the method will now check for LIVE stage, while the cube is in LIGHTING. But the cahUpdate method in FirstLightProcessor will return true. So essentially almost all skylight diffuse updates will be completely ignored, essentially falling back to Minecraft indef lighting for FirstLightProcessor.